File: bench.go
package main
import (
  "os"
  "bytes"
  "io/ioutil"
  "path/filepath"
  "github.com/PuerkitoBio/goquery"
  "log"
  "strings"
)

func main() {
  files_path := "/Users/sushi/Research/tie/html/test_files"
  files, _ := ioutil.ReadDir(files_path)
  for _, f := range files {
    where := filepath.Join(files_path, f.Name())
    file, _ := os.Open(where)
    doc, err := goquery.NewDocumentFromReader(file)
    if err != nil {
      log.Fatal(err)
    } else {
      var buf bytes.Buffer
      doc.Find("a").Each(func(i int, sel *goquery.Selection) {
        str, _ := goquery.OuterHtml(sel)
        buf.WriteString(str + "\n")
      });
      ioutil.WriteFile(strings.Replace(where, "test_files", "go_output", -1), buf.Bytes(), 0644)
    }
  }
}

File: bench.js
fs = require('fs');
cheerio = require('cheerio');
var path = "/Users/sushi/Research/tie/html/test_files"
fs.readdir(path, function(err, items) {
    for (var i=0; i<items.length; i++) {
      var fp = path + "/" + items[i];
      var file = fs.readFileSync(fp, "utf8");
      var $ = cheerio.load(file);
      fun_times = $.html("a");
      new_path = fp.replace("test_files", "node_output");
      fs.writeFileSync(new_path, fun_times, "utf8");
    }
});
